src/posix/client.cpp

```diff
@@ -316,7 +316,7 @@ int main(int argc, char *argv[])
 
                     if (buffer[i] == '\r' || buffer[i] == '\n')
                     {
-                        VerifyOrExit(DoWrite(STDOUT_FILENO, buffer + lineStart, i - lineStart + 1),
+                        VerifyOrExit(DoWrite(STDOUT_FILENO, buffer + lineStart, (size_t)(i - lineStart + 1)),
```

Comment:
Prefer C++-style casts:
```suggestion
                        VerifyOrExit(DoWrite(STDOUT_FILENO, buffer + lineStart, static_cast<size_t>(i - lineStart + 1)),
```